### PR TITLE
feat: allow configuration of repoUrl and revision (#59)

### PR DESCRIPTION
**Pull request by tback**
First PR here. Do you mind to let me know what's required to move this forward?

Signed-off-by: Till Backhaus <till@backha.us>

### CODE REVIEW

**Comment by jkroepke** [comment]
How to define a custom revision? Define a static revision as annotation may a bit pointless.

File: pkg/services/github.go [hunk 1]
```diff
@@ -31,49 +31,51 @@ type GitHubOptions struct {
 }
 
 type GitHubNotification struct {
```
Comment (pasha-codefresh) [inline pkg/services/github.go:33]: If you remove "Status" field it can cause breaking changes for existing customers, why do you need to change it?

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: pkg/services/github.go [hunk 2]
```diff
@@ -31,49 +31,51 @@ type GitHubOptions struct {
...
-	revisionTemplate = "{{.app.status.operationState.syncResult.revision}}"
+	defaultRepoURLtemplate  = "{{.app.spec.source.repoURL}}"
+	defaultRevisionTemplate = "{{.app.status.operationState.syncResult.revision}}"
 )
 
 func (g *GitHubNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+
+	repoURLtemplate := defaultRepoURLtemplate
+	if g.RepoURL != "" {
+		repoURLtemplate = g.RepoURL
```
Comment (nan-yu) [inline pkg/services/github.go:50]: I think `g.RepoURL` refers to the actual git repo URL (e.g. `https://github.com/argoproj-labs/argocd-notifications.git`), while `repoURLtemplate` refers to the API field path (e.g. `{{.app.spec.source.repoURL}}`).
I would suggest using two fields to separate them as #115.

**Comment by pasha-codefresh** [comment]
Solved already in https://github.com/argoproj/notifications-engine/pull/115  , Thank you for effort that you did!

**Pull request closed without merging**